sky/skylet/providers/gcp/config.py

```diff
@@ -807,6 +807,9 @@ def get_usable_vpc(config):
     ports = config["provider"].get("ports", [])
     user_rules = []
     for port in ports:
+        if port == 22:
+            # Not creating a duplicate rule for port 22
+            continue
```

Comment:
Question: should we handle this duplication in the upper level or just handle it during updating the firewalls / security groups for both clouds?
For example, what will happen if the user's existing firewalls already contains a rule for 8080, and in the task yaml, the user specifies `ports: 8080`. Will it cause issue?

Reply:
For GCP it will create a rule that only applies to the VM with a network tag equal to the cluster name, so it will have some duplicated rules but they will be cleaned up once the cluster is down. IIUC we could not check 22 for GCP and it should works fine as well.

Reply:
similar as the comments for `aws/config.py`, what if the user specified a port `0-1000`? This seems won't skip that port. Will that cause duplication issue?

Reply:
Yeah it will, but it should be fine for GCP to have some duplications. I'm thinking maybe we could just skip checking for GCP to make our code simpler since it won't trigger any error and will be automatically cleaned up when the cluster is down, which makes the code seem a little bit redundant. Wdyt..?

Reply:
If we decided to remove the code in GCP, we could remove it in Azure too since it also won't raise any errors (as long as the priority is not the same, which is guaranteed by our code) and will be cleaned up automatically.
<img width="1594" alt="image" src="https://github.com/skypilot-org/skypilot/assets/74357442/aeb883d2-b7a1-4acc-a43c-9dbaed630f8d">